apps/studio/src/components/sidebar/core/HiddenEntitiesModal.vue

```diff
@@ -146,13 +147,13 @@
         this.entities = [...this.hiddenEntities]
         this.schemas = [...this.hiddenSchemas]
       },
-      unhideSchema(index: number) { 
+      unhideSchema(index: number) {
         const [schema] = this.schemas.splice(index, 1)
-        this.$store.dispatch('hideEntities/removeSchema', schema)
+        this.trigger(AppEvent.toggleHideSchema, schema, false)
```

Comment:
Why fire an app event rather than just dispatching to the store?

Reply:
I just thought we could re-use of what we had in the `AppEvent.ts`, because we used to have `hideSchema` Event which now I change it to `toggleHideSchema`. (We can hide or show a schema with `toggleHideSchema` by passing a boolean)

In [VirtualTableList.vue](https://github.com/beekeeper-studio/beekeeper-studio/pull/1771/files#diff-095b36270d569f0d9e05acb75359243770919245870dddd4249dc5ba0a725296R134-R142), it's easier to track which item should I hide/show by this event because we know which schema it is immediately. The same thing with `togglePinTableList`.

Reply:
kk